### PR TITLE
time date fixes

### LINKED ISSUE #2438: Text and Image are Overlapping
On select of any name the image of user and name is overlapping. There is no space .

image

### PR DESCRIPTION
**Pull request by Gurmeenkour99**
Fixes #2438 #2360 #2245

### CODE REVIEW

File: app/Http/Controllers/Front/ClientController.php [hunk 1]
```diff
@@ -260,7 +260,7 @@ public function getInvoices()
                             }
                         })
                     ->addColumn('date', function ($model) {
-                        return getDateHtml($model->date);
+                        return getDateHtmlcopy($model->created_at);
                     })
```
Comment (sowmiyasubramaniyan) [inline app/Http/Controllers/Front/ClientController.php:264]: Do not change this from date to created_at. because In the admin panel, when creating a custom invoice, we have the option to set a specific date for the invoice. created_at is automatically generated.

File: resources/views/themes/default1/front/clients/show-invoice.blade.php [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <section class="box-content">
         @php
             $set = App\Model\Common\Setting::where('id', '1')->first();
-             $date = getDateHtml($invoice->date);
+             $date = getDateHtmlcopy($invoice->created_at);
```
Comment (sowmiyasubramaniyan) [inline resources/views/themes/default1/front/clients/show-invoice.blade.php:27]: Here too

File: resources/views/themes/default1/invoice/show.blade.php [hunk 3]
```diff
@@ -30,7 +30,7 @@
             <div class="col-12">
 
                 <?php $set = App\Model\Common\Setting::where('id', '1')->first(); 
-                $date = getDateHtml($invoice->date);
+                $date = getDateHtml($invoice->created_at);
```
Comment (sowmiyasubramaniyan) [inline resources/views/themes/default1/invoice/show.blade.php:33]: Here too

**Review by sowmiyasubramaniyan — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Front/ClientController.php (hunk 1 above)
Comment (Gurmeenkour99) [reply app/Http/Controllers/Front/ClientController.php:264]: Sure I will update this , Actually i've checked this and the time for selected date is not updating while selecting the particular date , i have checked in the initial DB records it is present but now it is 00.00.00 that means it is not saving the time now , also in development it is same so need to fix that issue before time miss match . Till then we can put the testing of this issue on hold.

File: app/Model/Order/Invoice.php [hunk 4]
```diff
@@ -94,8 +93,6 @@ public function getStatusAttribute($value)
     public function getCreatedAtAttribute($value)
     {
         $date1 = new DateTime($value);
-        $tz = \Auth::user()->timezone()->first()->name;
```
Comment (sandesh556) [inline app/Model/Order/Invoice.php:97]: why was this removed?

File: app/Model/Order/Order.php [hunk 5]
```diff
@@ -104,8 +103,6 @@ public function getOrderStatusAttribute($value)
     public function getCreatedAtAttribute($value)
     {
         $date1 = new DateTime($value);
-        $tz = \Auth::user()->timezone()->first()->name;
```
Comment (sandesh556) [inline app/Model/Order/Order.php:107]: why was this removed?

File: resources/views/themes/default1/front/clients/show-invoice.blade.php [hunk 6]
```diff
@@ -22,14 +22,6 @@
     <div class="featured-box featured-box-primary text-left mt-5" style="max-width: 900px">
 
     <section class="box-content">
-        @php
-            $set = App\Model\Common\Setting::where('id', '1')->first();
```
Comment (sandesh556) [inline resources/views/themes/default1/front/clients/show-invoice.blade.php:26]: why was this removed?

**Review by sandesh556 — CHANGES_REQUESTED** [review]

**Review by sandesh556 — DISMISSED** [review]

**Comment by bhanu2217** [comment]
**Feedback:**

- Ensure we have checked this functionality throughout the project and not missed any module/page

Looks good. Demo approved from my end.

**Review by sowmiyasubramaniyan — APPROVED** [review]

**Comment by Gurmeenkour99** [comment]
1. Date Time Mismatch
2.Text and image overlapping
3. Invoice number alignment on view of invoice .

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Last commit
image
Issue1:- In order data table still time is showing wrong.
image
Issue2:- User profile pic not showing in generate invoice page
image
Issue3:- In client panel order page still time is showing wrong.
image

**Comment by Gurmeenkour99** [comment]
Issue:1 and Issue:3 :It's the order execution time which is not same as invoice time . So it will pick the current time while executing not the invoice time .
Issue:2 : I have tried it on the same URL, there might be delay in loading time, please try once again .
image

**Comment by SarfrazS** [comment]
The user time zone is Australia/Melbourne 
After that, he purchased a product in the invoice popup message and the order data table showed his time zone time but in invoice data table showed system time.
image
order table
image
Invoice table
image

**Comment by bhanu2217** [comment]
**Some suggestions:** 
- Avoid using Jugaad
- Write this module at central/system level, all modules should be using it

**How it should work:**
- Any data saved into DB, Time zone to be converted from user time zone to UTC
- While displaying time to user, time zone to be converted from UTC to user time zone. 

Talk to @sandesh556 to figure out how to read users system(laptop, desktop) timezone.

**Comment by Gurmeenkour99** [comment]
@SarfrazS Please first set the local system timezone as the same you want your user to be , if you select your timezone different and the system timezone is different it will cause conflicts . If you choose to test via VPN please select the local timezone as same as VPN location.

**Comment by bhanu2217** [comment]
Many people today use VPNs.

System(laptop/desktop) Timezone should be independent of the VPN/IP address etc
Timezone Agora picks up should be of the system always and not as per VPN IP address or country.

**Comment by DhanyaChengappa** [comment]
@Gurmeenkour99 Please give a demo for this once work is completed and only then proceed
